src/classes/MyClass2.cls

```diff
@@ -1,4 +1,4 @@
-public with sharing class MyClass2 {
+public inherited sharing class MyClass2 {
```

Comment:
Class `MyClass2` does not enforce sharing correctly. Please change to `with sharing`.